properties/basic.py

```diff
@@ -764,21 +765,20 @@ def regex(self):
 
     @regex.setter
     def regex(self, value):
-        if not isinstance(value, string_types):
-            raise TypeError("'regex' must be a string")
-        try:
-            self._regex_compiled = re.compile(value)
-        except re.error:
-            raise ValueError("'regex' is not valid")
-        self._regex = value
+        if isinstance(value, string_types):
+            value = re.compile(value)
```

Comment:
Should we wrap this in a `try:`...`except:` as above to add useful context? I suppose I wonder if a random error from the `re` module is too vague... The errors from `re` are not really very clear.